File: springboot-backend/src/main/java/springboot/aviation/dto/response/FlightResponse.java
/* package springboot.aviation.dto.response;

import springboot.aviation.model.FlightStatus;
import springboot.aviation.model.Flight;

import static springboot.aviation.dto.utils.FormatUtils.formatDateTime;


public class FlightResponse {
    
    public Long id;
    public String flightNumber;
    public AirlineSummaryResponse airline;
    public AirportSummaryResponse departureAirport;
    public AirportSummaryResponse arrivalAirport;
    public String departureTime;
    public String arrivalTime;
    public FlightStatus status;

    private FlightResponse() {
    }

    public static FlightResponse from(Flight flight) {
        FlightResponse response = new FlightResponse();
        response.id = flight.hasId();
        response.flightNumber = flight.hasFlightNumber();
        response.airline = AirlineSummaryResponse.from(flight.hasAirline());
        response.departureAirport = AirportSummaryResponse.from(flight.hasDepartureAirport());
        response.arrivalAirport = AirportSummaryResponse.from(flight.hasArrivalAirport());
        response.departureTime = formatDateTime(flight.hasDepartureTime());
        response.arrivalTime = formatDateTime(flight.hasArrivalTime());
        response.status = flight.hasStatus();
        return response;
    }
}
 */

File: angular-frontend/src/app/flights/models/flight-response.ts
import { AirlineSummaryResponse } from './airline-summary-response';
import { AirportSummaryResponse } from './airport-summary-response'
import { FlightStatus } from './flight-status.enum';

export interface FlightResponse {
    id: number;
    flightNumber: string;
    airline: AirlineSummaryResponse;
    departureAirport: AirportSummaryResponse;
    arrivalAirport: AirportSummaryResponse;
    departureTime: string;
    arrivalTime: string;
    status: FlightStatus;
}
